plugins/auth-backend/src/service/router.ts

```diff
@@ -59,7 +60,9 @@ export async function createRouter({
   });
   const catalogApi = new CatalogClient({ discoveryApi: discovery });
 
-  router.use(cookieParser());
+  const secret = 'backstage secret'; // TODO: Allow an override here
+  router.use(cookieParser(secret));
+  router.use(session({ secret, saveUninitialized: false, resave: false }));
   router.use(express.urlencoded({ extended: false }));
   router.use(express.json());
 
```

Comment:
I missed this before, but there is as statically defined route on line 108 below:
```
  router.use(
    createOidcRouter({
      tokenIssuer,
      baseUrl: authUrl,
    }),
  );
```
This is implemented differently than the dynamic auth providers handled here.  It's not clear to me what the intention was with this statically defined route.

Reply:
This is actually part of the `BackstageIdentity` implementation. Still `WIP`, but essentially we're issuing OIDC ID Tokens as part of the sign-in process. The purpose of this endpoint is to allow for verification of those tokens by publishing the JWKs used to sign them. So related to auth but not what you'd be looking for for using an external OIDC IdP.

Reply:
I took a closer look, and it seems like the router is a stub for using Backstage itself as an OIDC endpoint.  If my understanding is correct, then this does not overlap with the functionality provided in this PR.